includes/class-wc-gateway-komoju-single-slug.php

```diff
@@ -25,9 +25,62 @@ public function __construct($payment_method)
             $this->icon = "https://komoju.com/payment_methods/$slug.svg";
         }
 
+        // TODO: It would be nice if KOMOJU told us in the payment method object whether or
+        // not it supports refunds. For now, we'll just wing it.
+        if (!in_array($slug, ['konbini', 'pay_easy', 'bank_transfer'])) {
+            $this->supports[] = 'refunds';
+        }
+
         parent::__construct();
     }
 
+    /**
+     * Process refund.
+     *
+     * Attempts to refund the passed-in amount with KOMOJU.
+     *
+     * @param int $order_id order ID
+     * @param float|null $amount refund amount
+     * @param string $reason refund reason
+     *
+     * @return bool true or false based on success, or a WP_Error object
+     */
+    public function process_refund($order_id, $amount = null, $reason = '')
+    {
+        $order      = wc_get_order($order_id);
+        $payment_id = $order->get_meta('komoju_payment_id');
+        $currency   = $order->get_currency();
+
+        if ($payment_id == '') {
+            return false;
+        }
+
+        $payload = [];
+        if (!is_null($amount)) {
+            $payload['amount'] = self::to_cents($amount, $currency);
+        }
+        if ($reason != '') {
+            $payload['description'] = $reason;
+        }
+
+        try {
+            $payment = $this->komoju_api->refund($payment_id, $payload);
+        } catch (KomojuExceptionBadServer | KomojuExceptionBadJson $e) {
+            $error_message = $e->getMessage();
+            $this->log($error_message);
+
+            return new WP_Error('komoju_refund_failed', $error_message);
+        }
+
+        $refund = $payment->refunds[count($payment->refunds) - 1];
+
+        if ($refund && $refund->amount == self::to_cents($amount, $currency)) {
```

Comment:
I am not familer with php but can you write something like this?
```php
return ($refund && $refund->amount == self::to_cents($amount, $currency)
```

Reply:
Sadly, PHP always spits out a bool for && 😭